tests/app/conftest.py

```diff
@@ -581,6 +581,25 @@ def sample_api_key(notify_db,
     return api_key
 
 
+@pytest.fixture(scope='function')
+def sample_service_data_api_key(service=None, key_type=KEY_TYPE_NORMAL, name=None):
+    if service is None:
+        service = create_service(check_if_service_exists=True)
+
+    data = {
+        'service': service,
+        'name': name or uuid.uuid4(),
+        'created_by': service.created_by,
+        'key_type': key_type
+    }
+    api_key = ApiKey(**data)
+    save_model_api_key(api_key)
+
+    _tmp = dao_fetch_service_by_id_with_api_keys(service.id)
+    service_data_api_key = _tmp.api_keys[0]
```

Comment:
This does exactly what we just avoided in the old `__create_token` method.

If `service` comes in as any service that already exists we run the risk of it already having an API key, and then we set `service_data_api_key` to that rather than the API key we just made because you're zero indexing it. 

With the way this is currently setup it will pass without checking the key you just created on line 595 if there is already a key for that service. Both this method and where this fixture is used index the services API keys' first key.

Please approach this in a more robust manner, but I think you can just delete lines 598-599 and change 600 to `return api_key`.